src/cpu/flexcpu/simple_dataflow_cpu.cc

```diff
@@ -388,17 +388,34 @@ SimpleDataflowCPU::requestMemRead(const RequestPtr& req, ThreadContext* tc,
 
     PacketPtr pkt = Packet::createRead(req);
     pkt->allocate();
+    // Note: pkt is deleted in DataPort::recvTimingResp(), except in the case
+    //       of mmappedIpr where it is deleted in the internally scheduled
+    //       event.
+
+    // Despite mmappedIpr accessing registers, we do not explicitly check any
+    // register dependencies under the assumption that the misc regs will not
+    // be accessed through other means, and that memory dependence resolution
+    // will be sufficient to ensure correct out-of-order behavior.
+    if (req->isMmappedIpr()) {
+        // TODO think about limiting access to mmapped IPR, either alongside
+        //      other memory requests, or as its own constraint
```

Comment:
FYI, I doubt it's going to be important to limit this. mmapped IPR should be very rare and likely mostly a startup thing.